crates/forge_domain/src/context.rs

```diff
@@ -14,6 +65,7 @@ use crate::{ToolChoice, ToolDefinition};
 pub enum ContextMessage {
     ContentMessage(ContentMessage),
     ToolMessage(ToolResult),
+    Attachments(HashSet<Attachment>),
```

Comment:
```suggestion
    Image {url: String},
```

Isn't this better? Image is a special message.

Reply:
shouldn't it be collection of images?